plasmapy/formulary/collisions.py

```diff
@@ -2033,3 +2038,157 @@ def coupling_parameter(
+            The value of the Coulomb logarithm evaluated for the two interacting particles.
+
+        Raises
+        ------
+        `ValueError`
+            If both :math:`v_a` and :math:`T_a` are specified, or neither
+            :math:`v_a` nor :math:`T_a` are specified.
+
+        Notes
+        -----
+        The collision frequencies between two interacting particles are given by four differential equations:
+
+            slowing down - :math:`\frac{d\textbf{v}_{α}}{dt}=-ν_{s}^{α\backslashβ}\textbf{v}_{α}`
+
+            transverse diffusion - :math:`\frac{d}{dt}\left(\overline{\textbf{v}}_{α}-\overline{\textbf{v}}_{α}\right)_{⊥}^{2}=ν_{⊥}^{α\backslashβ}v_{α}^{2}`
+
+            parallel diffusion - :math:`\frac{d}{dt}\left(\overline{\textbf{v}}_{α}-\overline{\textbf{v}}_{α}\right)_{∥}^{2}=ν_{∥}^{α\backslashβ}v_{α}^{2}`
+
+            energy loss - :math:`\frac{d}{dt}v_{α}^{2}=-ν_{ϵ}^{α\backslashβ}v_{α}^{2}`
+
+        For general solutions and limiting cases we encourage the curious reader to refer to p. 31 of :cite:t:`nrlformulary:2019`
+
+        This function uses CGS units under the hood to coincide with our references, but passing MKS units is acceptable.
+
+        Examples
+        --------
+        >>> import astropy.units as u
+        >>> T_a = 1 * u.eV
+        >>> n_b = 1e20 * u.cm**-3
+        >>> T_b = 1e3 * u.eV
+        >>> Coulomb_log = 10 * u.dimensionless_unscaled
+        >>> frequencies = CollisionFrequencies(
+        ...     "e-", "e-", T_a=T_a, n_b=n_b, T_b=T_b, Coulomb_log=Coulomb_log
+        ... )
+        >>> frequencies.slowing_down
+        <Quantity 1.83701432e+11 Hz>
+
+        See Also
+        --------
+        ~plasmapy.formulary.collisions.Coulomb_log : Evaluates the Coulomb
+            logarithm for two interacting electron species.
+
+        ~plasmapy.formulary.collisions.collision_frequency : The Lorentz collision frequency.
+        """
+
+        if v_a is None:
+            if T_a is not None:
+                v_a = thermal_speed(T_a, test_particle)  # .to(u.cm / u.s)
+            else:
+                raise ValueError("Please specify either v_a or T_a.")
+        elif T_a is not None:
+            raise ValueError("Please specify either v_a or T_a, not both.")
+
+        self.test_particle = test_particle
+        self.field_particle = field_particle
+        self.v_a = v_a
+        self.n_b = n_b
+        self.T_b = T_b
+        self.Coulomb_log = Coulomb_log
+
+        x = self._x()
+        phi = self._phi(x)
+        phi_prime = self._phi_prime(x)
+
+        mass_ratio = test_particle.mass / field_particle.mass
+
+        self.v_0 = self._v_0()
+        self.slowing_down = (1 + mass_ratio) * phi * self.v_0
+        self.transverse_diffusion = 2 * ((1 - 1 / (2 * x)) * phi + phi_prime) * self.v_0
+        self.parallel_diffusion = (phi / x) * self.v_0
+        self.energy_loss = 2 * (mass_ratio * phi - phi_prime) * self.v_0
+
+    def _v_0(self):
```

Comment:
Is this part equivalent to the Lorentz collision frequency in collision_frequency? If so, I recommend making this public, exposing it, and then planning to depreciate the `collision_frequency` function in a future release.

Reply:
How would I go about exposing it, besides just making it public? Just making sure to document that it's available within the class?

Reply:
Yeah, just remove the underscore so it's public and add documentation. I would give it a more descriptive name (`Lorentz_collision_frequency`?) and add a citation to Eq. 2.14 in  Callan to the Notes section for this method (as on the docs for the current collision_freq function)

Look through the existing `collision_freq` function and make sure tehre are no use-cases that aren't currently covered by this new implementation. If  you can cover all of them, I'd suggest we add a Depreciation warning to that function and open an issue to remove it a couple releases after this PR is merged.

@namurphy let me know if you have any objection to phasing out the older form of this function in favor of this new, more generalized class.

Reply:
Nope, no problem from me!